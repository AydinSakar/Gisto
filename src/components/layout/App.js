import React from 'react';
import PropTypes from 'prop-types';
import { ipcRenderer } from 'electron';
import styled from 'styled-components';
import * as Mousetrap from 'mousetrap';
import { withRouter } from 'react-router-dom';

import { baseAppColor } from 'constants/colors';

import SuperSearch from 'components/layout/SuperSearch';

const AppWrapper = styled.div`
  display: flex;
  background: #fff;
  flex-direction: column;
  justify-content: flex-start;
  height: 100vh;
  color: ${baseAppColor};
`;

export class App  extends React.Component {
  state = {
    showSuperSearch: false
  };

  componentDidMount() {
    Mousetrap.bind(['shift shift', 'ctrl+f'], this.toggleSuperSesrch);

    // Disallow to drop files
    document.addEventListener('dragover', (event) => event.preventDefault());
    document.addEventListener('dragenter', (event) => event.preventDefault());
    document.addEventListener('dragleave', (event) => event.preventDefault());
    document.addEventListener('drop', (event) => event.preventDefault());

    // Route if 'routeTo' event sent from main
    ipcRenderer.on('routeTo', (event, routeTo) => {
      this.props.history.push(routeTo);
    });
  }

  componentWillUnmount() {
    Mousetrap.unbind(['shift shift', 'ctrl+f'], this.toggleSuperSesrch);
  }

  toggleSuperSesrch = () => {
    this.setState((prevState) => ({
      showSuperSearch: !prevState.showSuperSearch
    }));
  };

  render() {
    return (
      <AppWrapper>
        { this.props.children }
        { this.state.showSuperSearch && (
          <SuperSearch toggleSuperSesrch={ () => this.toggleSuperSesrch() }/>
        ) }
      </AppWrapper>
    );
  }
}

App.propTypes = {
  children: PropTypes.node,
  history: PropTypes.object
};

export default withRouter(App);
