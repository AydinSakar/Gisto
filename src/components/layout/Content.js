import React from 'react';
import { HashRouter as Router, Route, Switch } from 'react-router-dom';
import styled from 'styled-components';
import { bg } from 'constants/colors';

import DashBoard from 'components/layout/content/DashBoard';
import Settings from 'components/layout/content/Settings';
import Snippet from 'components/layout/content/Snippet';
import About from 'components/layout/content/About';

const ContentWrapper = styled.div`
  flex: 1;
  padding: 20px;
  background: ${bg};
  overflow: auto;
`;

const Content = () => (
  <ContentWrapper>
    <Router>
      <Switch>
        <Route exact path="/" component={ DashBoard }/>
        <Route path="/about" component={ About }/>
        <Route path="/settings" component={ Settings }/>
        <Route path="/snippet/:id" component={ Snippet }/>
      </Switch>
    </Router>
  </ContentWrapper>
);

Content.propTypes = {};

export default Content;
