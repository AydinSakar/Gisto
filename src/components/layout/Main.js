import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { get } from 'lodash/fp';
import styled from 'styled-components';
import Sidebar from 'components/layout/sidebar/Sidebar';
import Content from 'components/layout/Content';

import * as snippetActions from 'actions/snippets';
import * as emojiActions from 'actions/emoji';
import { gaEvent } from 'utils/ga';

const MainWrapper = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  height: 100%;
`;

export class Main extends React.Component {
  componentDidMount() {
    const {
      getStarredSnippets, getSnippets, getEmoji
    } = this.props;

    getStarredSnippets();
    getSnippets();
    getEmoji();
    gaEvent({ category: 'general', action: 'App loaded' });
  }

  render() {
    const { edit, isCreateNew } = this.props;

    return (
      <MainWrapper>
        { !edit && !isCreateNew && <Sidebar/> }
        <Content/>
      </MainWrapper>
    );
  }
}

const mapStateToProps = (state) => ({
  edit: get(['ui', 'snippets', 'edit'], state),
  sinceLastUpdated: get(['snippets', 'snippets', 'lastUpdated'], state),
  isCreateNew: get(['router', 'location', 'pathname'], state) === '/new'
});

Main.propTypes = {
  edit: PropTypes.bool,
  isCreateNew: PropTypes.bool,
  getSnippets: PropTypes.func,
  getStarredSnippets: PropTypes.func,
  getEmoji: PropTypes.func
};

export default connect(mapStateToProps, {
  getSnippets: snippetActions.getSnippets,
  getEmoji: emojiActions.getEmoji,
  getStarredSnippets: snippetActions.getStarredSnippets
})(Main);
