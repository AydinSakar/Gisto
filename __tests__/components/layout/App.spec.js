import { App } from 'components/layout/App';

const propSetup = (props) => ({
  ...props
});

const setup = (props) => shallow(<App { ...propSetup(props) }/>);

describe('COMPONENTS - <App>', () => {
  test('render App', () => {
    const component = setup();

    expect(component).toMatchSnapshot();
  });
});
