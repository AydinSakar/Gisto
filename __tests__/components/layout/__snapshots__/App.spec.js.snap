// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENTS - <App> render App 1`] = `
<App>
  <ForwardRef>
    <StyledComponent
      forwardedClass={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "AppWrapper-sc-8qqffy-0",
            "isStatic": true,
            "lastClassName": "fJOKJi",
            "rules": Array [
              "display:flex;background:#fff;flex-direction:column;justify-content:flex-start;height:100vh;color:",
              "#3F84A8",
              ";",
            ],
          },
          "displayName": "AppWrapper",
          "render": [Function],
          "styledComponentId": "AppWrapper-sc-8qqffy-0",
          "target": "div",
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="AppWrapper-sc-8qqffy-0 fJOKJi"
      >
        <GlobalStyleComponent />
      </div>
    </StyledComponent>
  </ForwardRef>
</App>
`;
